pkg/middlewares/aws_s3_bucket_public_access_block_reconcilier.go

```diff
@@ -0,0 +1,80 @@
+package middlewares
+
+import (
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/sirupsen/logrus"
+	"github.com/snyk/driftctl/pkg/resource"
+	"github.com/snyk/driftctl/pkg/resource/aws"
+)
+
+// AwsS3BucketPublicAccessBlockReconciler middleware ignores every s3 bucket public block that is set to the default values (every option set to false)
+// This is used to avoid displaying false positive unmanaged resources.
+// The problem here is that the aws SDK can either return an error `NoSuchPublicAccessBlockConfiguration` while
+// retrieving bucket public block, or a response with all fields set to false (the default)
+//
+// To reproduce this edgy case you can do that:
+// - Disable this middleware
+// - Go to the folder of the test `TestAcc_Aws_S3Bucket_PublicAccessBlock` : `testdata/acc/aws_s3_bucket_public_access_block`
+// - Apply tf code
+// - Run a scan with the driftignore from the test folder (ignore everything but bucket and public access block)
+//   - Infra should be in sync (be sure that you have no dangling bucket in your aws test env)
+// - Create a new unmanaged bucket from the console, with every option from the policy block set to false
+// - Run the scan again
+//   - One resource should be unmanaged: the bucket (expected behavior)
+// - Go to the console and update public access block for that bucket
+// - Run the scan again
+//   - We should now have a new public access block resource unmanaged (expected)
+// - Now uncheck back all things in the public block you just updated
+// - Run the scan again
+//   - We still have the public block as unmanaged, this is NOT expected since all values are back to default
+//
+// This simple middleware is handling that edge case by removing resource that have every attribute set to false both
+// from resources from state and resources from remote.
```

Comment:
I have a problem with this middleware.

Imagine I want to use S3 as a static website, as per [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/website-hosting-custom-domain-walkthrough.html#root-domain-walkthrough-configure-bucket-permissions) I will set all Block Public Access settings to false just like below:

```
resource "aws_s3_bucket_public_access_block" "block" {
  bucket = aws_s3_bucket.test.id
  block_public_acls   = false
  block_public_policy = false
  ignore_public_acls = false
  restrict_public_buckets = false
}
```

With this middleware, each driftctl scan will by design ignore my managed resource. And thus, if someone tries to change a setting in the console, instead of having a drifted managed resource I will have a new unmanaged resource which I'm no expecting.

```
# change manually in the console and run driftctl scan again
Found resources not covered by IaC:
  aws_s3_bucket_public_access_block:
    - test-bucket-driftctl-snyk
```

Reply:
That's true, good catch I missed that one.
Maybe we can try to find an according block in IaC when iterating on remote ones, and we can remove it only if we get a match ? That could handle that case WDYT ?

Reply:
I don't understand your thinking here, can you try with an example?